exercises/practice/bob/.approaches/if/content.md

```diff
@@ -0,0 +1,60 @@
+# `if` statements
+
+```csharp
+    // uses System.Linq;
+    public static string Response(string message)
+    {
+        if (message.IsSilence())
+            return "Fine. Be that way!";
+
+        if (message.IsYell() && message.IsQuestion())
+            return "Calm down, I know what I'm doing!";
+
+        if (message.IsYell())
+            return "Whoa, chill out!";
+
+        if (message.IsQuestion())
+            return "Sure.";
+
+        return "Whatever.";
+    }
+
+    private static bool IsSilence(this string message) =>
+        string.IsNullOrWhiteSpace(message);
+
+    private static bool IsYell(this string message) =>
+        message.Any(char.IsLetter) && message.ToUpperInvariant() == message;
+
+    private static bool IsQuestion(this string message) =>
+        message.TrimEnd().EndsWith("?");
+```
+
+In this approach you add behavior to the String type using [Extension methods][extension-methods].
+A series of `if` statements use the added methods to evaluate the conditions.
+As soon as the right condition is found, the correct response is returned.
+
+Note that there are no `else if` or `else` statements.
+If an `if` statement can return, then an `else if` or `else` is not needed.
+Execution will either return or will continue to the next statement anyway.
+
+The [LINQ][linq] method [EndsWith][endswith] is used to determine if the input ends with a question mark.
+
+The `String` method [IsNullOrWhiteSpace][isnullorwhitespace] is used to determine if the input is `null` or only contains [whitespace][whitespace] characters.
+
+The first half of the yell condition is constructed from the [Any][any] LINQ method and the [IsLetter][isletter] `Char` method to ensure there is at least one letter character in the `String`.
+This is because the second half of the condition tests that the uppercased input is the same as the input.
+If the input were only `"123"` it would equal itself uppercased, but without letters it would not be a yell.
+The uppercasing is done by using the `String` method [ToUpperInvariant][toupperinvariant].
+The invariant culture represents a culture that is culture-insensitive.
+It is associated with the English language but not with a specific country or region.
+For more information, see the [CultureInfo.InvariantCulture][invariantculture] property.
```

Comment:
Oh, I seem to remember that links don't work within exercism/XXX blocks, so I'll put the link for more info just below the note.

Reply:
Well, they don't if they use the reference link style, but they do if the link is inlined 👼

Reply:
Ahhh, yeah, I probably used the reference style, although I would have thought I tried the inline style before abandoning the attempt, but I probably didn't, since you don't really know how it will work until the file is published.